src/tests/ov_helpers/ov_snippets_models/src/subgraph_mha.cpp

```diff
@@ -81,60 +109,107 @@ std::shared_ptr<ov::Model> MHAFunction::initReference() const {
-        auto mulConst = ngraph::builder::makeConstant(precisions[1], constantShapes[2], mulConstData, true);
-        auto mulParam = std::make_shared<ngraph::opset1::Parameter>(precisions[1], mulConst->get_shape());
+        ov::Shape shape(rank, 1);
+        shape[rank - 3] = transpose1->get_output_shape(0)[rank - 3];
+        std::vector<float> mulConstData(ngraph::shape_size(shape));
+        const auto mulConst = ngraph::builder::makeConstant(precisions[1], shape, mulConstData, true);
+        const auto mulParam = std::make_shared<ngraph::opset1::Parameter>(precisions[1], mulConst->get_shape());
         matmul_parent1 = std::make_shared<ngraph::opset3::Multiply>(transpose1, mulParam);
         subgraph_params = {transpose0Param, transpose1Param, mulParam, addParam, transpose2Param};
         subgraph_inputs = {data0, data1, mulConst, data2, data3};
     }
-    const auto matMul0 = std::make_shared<ngraph::opset3::MatMul>(transpose0, matmul_parent1, transA, transB);
+
+    const auto matMul0 = std::make_shared<ngraph::opset3::MatMul>(transpose0, matmul_parent1);
     const auto add = std::make_shared<ngraph::opset3::Add>(matMul0, addParam);
-    const auto reshape0 = std::make_shared<ngraph::opset1::Reshape>(add, reshape0Const, true);
-    const auto softMax = std::make_shared<ngraph::opset1::Softmax>(reshape0, 1);
-    const auto reshape1 = std::make_shared<ngraph::opset1::Reshape>(softMax, reshape1Const, true);
+    const auto softMax = std::make_shared<ngraph::opset1::Softmax>(add, rank - 1);
     const auto transpose2 = std::make_shared<ov::op::v1::Transpose>(transpose2Param, transpose2Const);
-    const auto matMul1 = std::make_shared<ngraph::opset3::MatMul>(reshape1, transpose2, transA, transB);
+    const auto matMul1 = std::make_shared<ngraph::opset3::MatMul>(softMax, transpose2);
     const auto transpose3 = std::make_shared<ov::op::v1::Transpose>(matMul1, transpose3Const);
 
     auto subgraph = std::make_shared<ov::snippets::op::Subgraph>(subgraph_inputs,
             std::make_shared<ov::Model>(NodeVector{transpose3}, subgraph_params));
 
     return std::make_shared<ov::Model>(NodeVector{subgraph}, ngraphParams);
 }
+std::shared_ptr<ov::Model> MHASplitMFunction::initReference() const {
+    auto data0 = std::make_shared<ngraph::opset1::Parameter>(precisions[0], input_shapes[0]);
+    auto data1 = std::make_shared<ngraph::opset1::Parameter>(precisions[1], input_shapes[1]);
+    auto data2 = std::make_shared<ngraph::opset1::Parameter>(precisions[2], input_shapes[2]);
+    auto data3 = std::make_shared<ngraph::opset1::Parameter>(precisions[3], input_shapes[3]);
+    ngraph::ParameterVector ngraphParams = {data0, data1, data2, data3};
+
+    auto make_reshape = [](const std::shared_ptr<ov::Node>& node, const ov::Shape& new_shape) {
+        auto shape_const = ngraph::builder::makeConstant(ngraph::element::i32, {new_shape.size()}, new_shape);
+        return std::make_shared<ov::op::v1::Reshape>(node, shape_const, true);
+    };
```

Comment:
FYI: it looks like [reshapeTo](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/utils/utils.cpp#L73) does the same